src/cwt.rs

```diff
@@ -0,0 +1,128 @@
+use std::collections::BTreeMap;
+use std::ops::Deref;
+
+pub use crate::claim::{self, Claim, Label, NumericDate};
+
+/// CWT claims set.
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq, Eq)]
+pub struct ClaimsSet(BTreeMap<Value, Value>);
+
+impl ClaimsSet {
+    pub fn insert_claim<T: Claim>(&mut self, claim: T) -> Option<Value> {
+        self.0.insert(T::label().into(), claim.into())
+    }
+
+    pub fn get_claim<T: Claim>(&self) -> Option<&Value> {
+        self.0.get(&T::label().into())
+    }
+
+    pub fn serialize(&self) -> Result<Vec<u8>, serde_cbor::Error> {
+        serde_cbor::to_vec(&self)
+    }
+}
+
+impl AsRef<BTreeMap<Value, Value>> for ClaimsSet {
+    fn as_ref(&self) -> &BTreeMap<Value, Value> {
+        &self.0
+    }
+}
+
+impl Deref for ClaimsSet {
+    type Target = BTreeMap<Value, Value>;
+
+    fn deref(&self) -> &BTreeMap<Value, Value> {
+        &self.0
+    }
+}
+
+impl From<BTreeMap<Value, Value>> for ClaimsSet {
+    fn from(m: BTreeMap<Value, Value>) -> Self {
+        Self(m)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::define_claim;
+
+    use super::*;
+
+    fn test_cases() -> Vec<(&'static str, ClaimsSet, Vec<u8>)> {
+        // Checked against coset
+        let serialized1 = hex::decode("a70175636f61703a2f2f61732e6578616d706c652e636f6d02656572696b77037818636f61703a2f2f6c696768742e6578616d706c652e636f6d041a5612aeb005fb41d584367c200000060007420b71").unwrap();
+        let mut claims_set1 = ClaimsSet::default();
+        // Basic test case tweaks RFC8392 example to include fractional & "0" dates
+        claims_set1.insert_claim(claim::Issuer::new("coap://as.example.com".into()));
+        claims_set1.insert_claim(claim::Subject::new("erikw".into()));
+        claims_set1.insert_claim(claim::Audience::new("coap://light.example.com".into()));
+        claims_set1.insert_claim(claim::ExpirationTime::new(NumericDate::Integer(1444064944)));
+        claims_set1.insert_claim(claim::NotBefore::new(NumericDate::Fractional(1443944944.5)));
+        claims_set1.insert_claim(claim::IssuedAt::new(NumericDate::Integer(0)));
+        claims_set1.insert_claim(claim::CWTId::new(hex::decode("0b71").unwrap()));
+
+        // Reordered case above
+        let mut claims_set2 = ClaimsSet::default();
+        claims_set2.insert_claim(claim::IssuedAt::new(NumericDate::Integer(0)));
+        claims_set2.insert_claim(claim::Subject::new("erikw".into()));
+        claims_set2.insert_claim(claim::NotBefore::new(NumericDate::Fractional(1443944944.5)));
+        claims_set2.insert_claim(claim::CWTId::new(hex::decode("0b71").unwrap()));
+        claims_set2.insert_claim(claim::ExpirationTime::new(NumericDate::Integer(1444064944)));
+        claims_set2.insert_claim(claim::Audience::new("coap://light.example.com".into()));
+        claims_set2.insert_claim(claim::Issuer::new("coap://as.example.com".into()));
+
+        // Create some dummy claims using negative/text labels
+        define_claim!(
```

Comment:
let me know if this is too strange (+ the macro overall -- happy to drop the macro entirely (feel free to take a look at the commit `a4045092e0ed07e58aabf978ee3323b6df4f576e` before the macro change to see how that is in comparison, or if you have any other ideas)

Reply:
The macro is fine for internal use, but I don't think we should make it visible in the API